src/CommunityToolkit/Xamarin.CommunityToolkit/Helpers/CommandFactory.shared.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Threading.Tasks;
+using Xamarin.CommunityToolkit.ObjectModel;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Helpers
+{
+	public static class CommandFactory
+	{
+		public static Command Create(Action execute, Func<bool> canExecute = null) =>
+			new Command(execute, canExecute ?? (() => true));
+
+		public static Command Create<TExecute>(Action<TExecute> execute, Func<bool> canExecute = null) =>
+			new Command(ConvertExecute(execute), ConvertCanExecute(canExecute));
+
+		public static Command Create<TExecute, TCanExecute>(Action<TExecute> execute, Func<TCanExecute, bool> canExecute) =>
+			new Command(ConvertExecute(execute), ConvertCanExecute(canExecute));
+
+		public static IAsyncCommand Create(
+			Func<Task> execute,
+			Func<bool> canExecute = null,
+			Action<Exception> onException = null,
+			bool continueOnCapturedContext = false,
+			bool allowsMultipleExecutions = true) =>
+			new AsyncCommand(execute, canExecute, onException, continueOnCapturedContext, allowsMultipleExecutions);
+
+		public static IAsyncCommand<TExecute> Create<TExecute>(
+			Func<TExecute, Task> execute,
+			Func<bool> canExecute = null,
+			Action<Exception> onException = null,
+			bool continueOnCapturedContext = false,
+			bool allowsMultipleExecutions = true) =>
+			new AsyncCommand<TExecute>(execute, canExecute, onException, continueOnCapturedContext, allowsMultipleExecutions);
+
+		public static IAsyncCommand<TExecute, TCanExecute> Create<TExecute, TCanExecute>(
+			Func<TExecute, Task> execute,
+			Func<TCanExecute, bool> canExecute = null,
+			Action<Exception> onException = null,
+			bool continueOnCapturedContext = false,
+			bool allowsMultipleExecutions = true) =>
+			new AsyncCommand<TExecute, TCanExecute>(execute, canExecute, onException, continueOnCapturedContext, allowsMultipleExecutions);
+
+		static Action<object> ConvertExecute<T>(Action<T> execute)
+		{
+			if (execute is null)
```

Comment:
we don't use pattern matching for null check, can you fix this one and others?
```suggestion
			if (execute == null)
```